front/app/components/Form/InputControl.tsx

```diff
@@ -37,21 +38,26 @@ const InputControl = ({
   return (
     <>
       <FormLabel
-        htmlFor={id}
+        htmlFor={sanitizeForClassname(id)}
         labelValue={getLabel(uischema, schema, path)}
         optional={!required}
         subtextValue={schema.description}
         subtextSupportsHtml
       />
       <Input
-        id={id}
+        id={sanitizeForClassname(id)}
         type={schema.type === 'number' ? 'number' : 'text'}
         value={data}
         onChange={onChange}
         maxCharCount={schema?.maxLength}
-        onBlur={() => setDidBlur(true)}
+        onBlur={() => {
+          uischema?.options?.transform === 'trim_on_blur' &&
+            isString(data) &&
+            onChange(data.trim());
```

Comment:
We discussed already but for transparency : we didn't but we need it since these fields are often used with a minimal or maximal length or are simply required, and BE trims those already, making the behaviour inconsistent between FE validation and BE validation. (Today if you type only spaces in the idea form title, the front-edn validation doesn't complain and the back-end returns 422 title can't be blank, which is not handled by the front-end at all and the button remains foreven loading.)